test/test_torch.py

```diff
@@ -6021,6 +6021,73 @@ def test_pow(self, device):
             torch.pow(m1, 1, out=out)
             self.assertEqual(out, m1)
 
+    @dtypes(*list(product(torch.testing.get_all_dtypes(include_bool=False),
+                          torch.testing.get_all_dtypes(include_bool=False))))
+    def test_float_power(self, device, dtypes):
+        base_dtype = dtypes[0]
+        exp_dtype = dtypes[1]
+        if base_dtype.is_complex or exp_dtype.is_complex:
+            dtype = torch.complex128
+        else:
+            dtype = torch.float64
+
+        base = torch.randint(-100, 100, (30,), device=device).to(dtype=base_dtype)
+        base[0] = base[8] = base[9] = 0
+        exp = torch.randint(-100, 100, (30,), device=device).to(dtype=exp_dtype)
```

Comment:
When testing the variants (below) enumerate an iterable exps that includes this tensor as well as the scalar values (-2, -1, 0, 1, 2).

Reply:
Addressed!